### PR TITLE
Interface changes related to CLOUDSTACK-8580

### PR DESCRIPTION
**Pull request by borisroman**
See issue CLOUDSTACK-8580 and individual commits.

### CODE REVIEW

**Comment by borisroman** [comment]
@wilderrodrigues @DaanHoogland @remibergsma Here are the interface changes related to #593.

File: ui/scripts/instances.js [hunk 1]
```diff
@@ -618,11 +618,11 @@
...
                             title: 'label.action.destroy.instance',
                             desc: 'label.action.destroy.instance',
-                isWarning: true,
+                            isWarning: true,
                             preFilter: function(args) {
-                                if (isAdmin() || isDomainAdmin()) {
-                                    args.$form.find('.form-item[rel=expunge]').css('display', 'inline-block');
-                                } else {
+                                // Hide the expunge checkbox when the authenticated user
+                                // can't expunge VMs. Related to CLOUDSTACK-8580.
```
Comment (DaanHoogland) [inline ui/scripts/instances.js]: I think the var name is explanatory. the comment could go with allocation and not use of the thingy.

File: ui/scripts/instances.js [hunk 2]
```diff
@@ -2425,11 +2425,15 @@
         var allowedActions = [];
 
         if (jsonObj.state == 'Destroyed') {
-            if (isAdmin() || isDomainAdmin()) {
+            // Display expunge and recover action when authenticated user
+            // is allowed to expunge or recover VMs. Related to CLOUDSTACK-8580.
```
Comment (DaanHoogland) [inline ui/scripts/instances.js]: see comment at line 624

File: ui/scripts/instances.js [hunk 3]
```diff
@@ -2498,8 +2502,11 @@
         } else if (jsonObj.state == 'Error') {
             allowedActions.push("destroy");
         } else if (jsonObj.state == 'Expunging') {
-            if (isAdmin() || isDomainAdmin())
+            // Display expunge action when authenticated user
+            // is allowed to expunge VMs. Related to CLOUDSTACK-8580.
+            if (g_allowUserExpungeRecoverVm) {
```
Comment (DaanHoogland) [inline ui/scripts/instances.js]: see commetn at line 624

**Comment by DaanHoogland** [comment]
please comment on testing the stuff, for instance ref unit - or integration tests that cover the code in the PR description/comment or, alternatively add unit tests.

**Comment by remibergsma** [comment]
@borisroman Please squash the commits here as well

**Comment by borisroman** [comment]
@remibergsma Done.

**Comment by wido** [comment]
Code looks good to me, but I see there are still some outstanding comments?

**Comment by DaanHoogland** [comment]
@wido my comments we're on outdated code. I will have another look

File: server/src/com/cloud/server/ManagementServerImpl.java [hunk 4]
```diff
@@ -3378,6 +3379,10 @@ private String signRequest(final String request, final String key) {
         final Integer apiLimitInterval = Integer.valueOf(_configDao.getValue(Config.ApiLimitInterval.key()));
         final Integer apiLimitMax = Integer.valueOf(_configDao.getValue(Config.ApiLimitMax.key()));
 
+        // Check if users can view destroyed vm's, expunge and/or recover them. Admins can always do so.
```
Comment (DaanHoogland) [inline server/src/com/cloud/server/ManagementServerImpl.java]: not really a comment that enlightens us beyond the naming of the variables, so could be omitted.

File: ui/scripts/cloudStack.js [hunk 5]
```diff
@@ -143,8 +143,11 @@
                         if (json.listcapabilitiesresponse.capability.userpublictemplateenabled != null) {
                             g_userPublicTemplateEnabled = json.listcapabilitiesresponse.capability.userpublictemplateenabled.toString(); //convert boolean to string if it's boolean
                         }
-                        g_userProjectsEnabled = json.listcapabilitiesresponse.capability.allowusercreateprojects;
 
+                        // Hide the expunge/recover checkboxes when the authenticated user
```
Comment (DaanHoogland) [inline ui/scripts/cloudStack.js]: this comment is misleading. the hiding is not done here

**Comment by DaanHoogland** [comment]
code LGTM, less comment is better in this case but I'll settle.

**Comment by borisroman** [comment]
@wido @DaanHoogland Removed last comments.

**Comment by remibergsma** [comment]
@borisroman Great, looks like we're ready to go. I'll wait for Travis to turn green and then merge.

**Pull request merged**